### ISSUE TITLE
Issue with floating-bits-0.3.0.0 when building

### ISSUE THREAD

**Issue by AnthonyDario** [nasa/PRECiSA#9]
Hello, I am having some trouble building PRECiSA. Specifically the `cabal -v2-install` step. I'm getting a type error in one of the dependencies (floating-bits-0.3.0.0).

Have you seen this before?  I was wondering if a dependency in `precisa.cabal` has been updated since this was last built.

```
$ cabal -wghc-9.2.8 v2-install
Wrote tarball sdist to
/Users/ardario/libs/PRECiSA/PRECiSA/dist-newstyle/sdist/precisa-3.2.1.tar.gz
Resolving dependencies...
Build profile: -w ghc-9.2.8 -O1
In order, the following will be built (use -v for more details):
 - directory-1.3.8.1 (lib:directory) (requires build)
 - floating-bits-0.3.0.0 (lib) (requires build)
 - ansi-terminal-1.0 (lib) (requires build)
 - scientific-0.3.7.0 (lib) (requires build)
 - process-1.6.17.0 (lib:process) (requires build)
 - prettyprinter-ansi-terminal-1.1.3 (lib) (requires build)
 - extra-1.7.14 (lib) (requires build)
 - optparse-applicative-0.18.1.0 (lib) (requires build)
 - precisa-3.2.1 (lib) (requires build)
 - precisa-3.2.1 (exe:precisa) (requires build)
Starting     floating-bits-0.3.0.0 (lib)
Starting     scientific-0.3.7.0 (lib)
Starting     ansi-terminal-1.0 (lib)
Starting     directory-1.3.8.1 (all, legacy fallback)
Building     floating-bits-0.3.0.0 (lib)
Building     scientific-0.3.7.0 (lib)
Building     ansi-terminal-1.0 (lib)
Installing   ansi-terminal-1.0 (lib)
Completed    ansi-terminal-1.0 (lib)
Building     directory-1.3.8.1 (all, legacy fallback)
Installing   scientific-0.3.7.0 (lib)
Completed    scientific-0.3.7.0 (lib)
Installing   directory-1.3.8.1 (all, legacy fallback)
Completed    directory-1.3.8.1 (all, legacy fallback)

Failed to build floating-bits-0.3.0.0.
Build log (
/Users/ardario/.cabal/logs/ghc-9.2.8/fltng-bts-0.3.0.0-67871e9e.log ):
Configuring library for floating-bits-0.3.0.0..
Warning: 'ghc-prof-options: -prof' is not necessary and will lead to problems
when used on a library. Use the configure flag --enable-library-profiling
and/or --enable-profiling.
Preprocessing library for floating-bits-0.3.0.0..
Building library for floating-bits-0.3.0.0..
[1 of 3] Compiling Data.Bits.Floating.Prim ( src/Data/Bits/Floating/Prim.hs, dist/build/Data/Bits/Floating/Prim.o, dist/build/Data/Bits/Floating/Prim.dyn_o )

src/Data/Bits/Floating/Prim.hs:58:34: error:
    • Couldn't match expected type ‘Word32#’ with actual type ‘Word#’
    • In the first argument of ‘W32#’, namely ‘(float2WordBitwise# f)’
      In the expression: W32# (float2WordBitwise# f)
      In an equation for ‘float2WordBitwise’:
          float2WordBitwise (F# f) = W32# (float2WordBitwise# f)
   |
58 | float2WordBitwise (F# f) = W32# (float2WordBitwise# f)
   |                                  ^^^^^^^^^^^^^^^^^^^^

src/Data/Bits/Floating/Prim.hs:63:53: error:
    • Couldn't match expected type ‘Word#’ with actual type ‘Word32#’
    • In the first argument of ‘word2FloatBitwise#’, namely ‘w’
      In the first argument of ‘F#’, namely ‘(word2FloatBitwise# w)’
      In the expression: F# (word2FloatBitwise# w)
   |
63 | word2FloatBitwise (W32# w) = F# (word2FloatBitwise# w)
   |                                                     ^
cabal: Failed to build floating-bits-0.3.0.0 (which is required by exe:precisa
from precisa-3.2.1). See the build log above for details.
```

**Comment by AnthonyDario** [comment]
I seem to have solved the problem by using ghc version 9.0.2. It seems that the change from base-4.15 to base-4.16 changed the way `Word#` worked, as documented [here](https://gitlab.haskell.org/ghc/ghc/-/wikis/migration/9.2#sized-wordint-use-sized-wordint-primitives).  I have no particular reason for selecting version 9.0.2 other than it's the latest ghc that doesn't use base 4.16 or greater.

**Comment by lauratitolo** [comment]
Thank you, Anthony, I will look into this.
